torch/_refs/__init__.py

```diff
@@ -797,15 +798,18 @@ def isclose(
     atol: float = 1e-08,
     equal_nan: bool = False,
 ) -> TensorLikeType:
-    if a.dtype != b.dtype:
-        msg = "Attempting to compare tensors of different dtypes {0} and {1}!".format(
-            a.dtype, b.dtype
-        )
-        raise ValueError(a, b)
-    if rtol < 0:
-        msg = "rtol must be greater than or equal to zero, but got {0}!".format(rtol)
-    if atol < 0:
-        msg = "atol must be greater than or equal to zero, but got {0}!".format(atol)
+    check_value(
+        a.dtype == b.dtype,
+        lambda: f"Attempting to compare tensors of different dtypes {a.dtype} and {b.dtype}!",
```

Comment:
Nice update

Note to us: we should probably update `check_value` so it can just accept a string, too